File: src/app/candles-page/Product.ts
export abstract class Product {
    private id: string;
    private productType: string;
    private description: string;
    private stock: number;
    private price: number;
    private created: Date;

    constructor(id: string, productType: string, description: string, stock: number, price: number, created: Date) {
        this.id = id;
        this.productType = productType;
        this.description = description;
        this.stock = stock;
        this.price = price;
        this.created = created;
    }
}

File: backend/spring-boot/backend/src/main/java/com/darkcandles/backend/products/Product.java
package com.darkcandles.backend.products;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document
@NoArgsConstructor
public abstract class Product {
  @Id
  private String id;
  private ProductType productType;
  private String description;
  private Integer stock;
  private Double price;
  private LocalDateTime created;

  public Product(ProductType productType, String description, Integer stock, Double price, LocalDateTime created) {
    this.productType = productType;
    this.description = description;
    this.stock = stock;
    this.price = price;
    this.created = created;
  }
}
